langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/FunctionCallHelper.java

```diff
@@ -0,0 +1,130 @@
+        //TODO: is it covering all the types correctly?
+        switch (type) {
+            case "string":
+                return Type.STRING;
+            case "integer":
+                return Type.INTEGER;
+            case "boolean":
+                return Type.BOOLEAN;
+            case "number":
+                return Type.NUMBER;
+            case "array":
+                return Type.ARRAY;
+            case "object":
+                return Type.OBJECT;
+            default:
+                return Type.TYPE_UNSPECIFIED;
+        }
+    }
+
+    static FunctionCall fromToolExecutionRequest(ToolExecutionRequest toolExecutionRequest) {
+        FunctionCall.Builder fnCallBuilder = FunctionCall.newBuilder()
+            .setName(toolExecutionRequest.name());
+
+        Struct.Builder structBuilder = Struct.newBuilder();
+        try {
+            JsonFormat.parser().merge(toolExecutionRequest.arguments(), structBuilder);
+        } catch (InvalidProtocolBufferException e) {
+            throw new RuntimeException(e);
+        }
+        Struct argsStruct = structBuilder.build();
+        fnCallBuilder.setArgs(argsStruct);
+
+        return fnCallBuilder.build();
+    }
+
+    static ToolExecutionRequest fromFunctionCall(FunctionCall functionCall) {
+        ToolExecutionRequest.Builder builder = ToolExecutionRequest.builder()
+            .name(functionCall.getName());
+
+        //TODO: can we make the mapping more transparently/automatically?
+
+        Map<String, Object> callArgsMap = new HashMap<>();
+        Struct callArgs = functionCall.getArgs();
+        Map<String, Value> callArgsFieldsMap = callArgs.getFieldsMap();
+        callArgsFieldsMap.forEach((key, value) -> callArgsMap.put(key, unwrapProtoValue(value)));
+
+        String serializedArgsMap = ToolExecutionRequestUtil.GSON.toJson(callArgsMap);
+        builder.arguments(serializedArgsMap);
+
+        return builder.build();
+    }
+
+    static Object unwrapProtoValue(Value value) {
+        Object unwrappedValue;
+        switch (value.getKindCase()) {
+            case NUMBER_VALUE:
+                unwrappedValue = value.getNumberValue();
+                break;
+            case STRING_VALUE:
+                unwrappedValue = value.getStringValue();
+                break;
+            case BOOL_VALUE:
+                unwrappedValue = value.getBoolValue();
+                break;
+            case STRUCT_VALUE:
+                HashMap<String, Object> mapForStruct = new HashMap<>();
+                value.getStructValue().getFieldsMap().forEach((key, val) -> mapForStruct.put(key, unwrapProtoValue(val)));
+                unwrappedValue = mapForStruct;
+                break;
+            case LIST_VALUE:
+                unwrappedValue = value.getListValue().getValuesList().stream().map(FunctionCallHelper::unwrapProtoValue).collect(Collectors.toList());
+                break;
+            default: // NULL_VALUE, KIND_NOT_SET, and default
```

Comment:
To be honest, I don't know when the `KIND_NOT_SET` case ever happens, so I don't know what kind of value it could contain, and I couldn't even find how to retrieve that value, so that's why I've just returned `null`.
Normally, I don't think this case happens in the Gemini API.